sdk/storage/azure-storage-blob/tests/test_retry_async.py

```diff
@@ -527,6 +530,38 @@ async def test_streaming_retry(self, **kwargs):
             count = [0]
             blob._pipeline._transport.send = self._count_wrapper(count, blob._pipeline._transport.send)
             await blob.download_blob()
-        assert stream_reader_read_mock.call_count == count[0] == 4
+        assert stream_reader_read_mock.call_count == count[0] == 12
+
+    @pytest.mark.live_test_only
+    @BlobPreparer()
+    async def test_streaming_retry_with_timeout(self, **kwargs):
+        storage_account_name = kwargs.pop("storage_account_name")
+        storage_account_key = kwargs.pop("storage_account_key")
+
+        """Test that retry mechanisms are working when streaming data."""
+        container_name = self.get_resource_name('utcontainer')
+        retry = LinearRetry(backoff=0.1, random_jitter_range=0)
+
+        service = self._create_storage_service(
+            BlobServiceClient, storage_account_name, storage_account_key, retry_policy=retry)
+        container = service.get_container_client(container_name)
+        try:
+            await container.create_container()
+        except ResourceExistsError:
+            pass
+        assert await container.exists()
+        blob_name = "myblob"
+        await container.upload_blob(blob_name, b"abcde", overwrite=True)
+
+        stream_reader_read_mock = mock.MagicMock()
+        future = asyncio.Future()
+        future.set_exception(ServerTimeoutError())
+        stream_reader_read_mock.return_value = future
+        with mock.patch.object(StreamReader, "read", stream_reader_read_mock), pytest.raises(HttpResponseError):
+            blob = container.get_blob_client(blob=blob_name)
+            count = [0]
+            blob._pipeline._transport.send = self._count_wrapper(count, blob._pipeline._transport.send)
+            await blob.download_blob()
+        assert stream_reader_read_mock.call_count == count[0] == 12
```

Comment:
Same reasoning for the 12 here. SO, now that we are properly raising the exception, should we revert the retry logic (but just expand the net for what exceptions are caught in `_download_async.py`